ldes-server-retention/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/services/retentionpolicy/definition/timeandversionbased/TimeAndVersionBasedRetentionPolicy.java

```diff
@@ -0,0 +1,50 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.retention.services.retentionpolicy.definition.timeandversionbased;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.retention.services.retentionpolicy.definition.RetentionPolicy;
+import be.vlaanderen.informatievlaanderen.ldes.server.retention.services.retentionpolicy.definition.RetentionPolicyType;
+import be.vlaanderen.informatievlaanderen.ldes.server.retention.services.retentionpolicy.definition.timebased.TimeBasedRetentionPolicy;
+import be.vlaanderen.informatievlaanderen.ldes.server.retention.services.retentionpolicy.definition.versionbased.VersionBasedRetentionPolicy;
+
+import java.time.Duration;
+
+import static be.vlaanderen.informatievlaanderen.ldes.server.retention.services.retentionpolicy.creation.RetentionPolicyConstants.TIME_BASED_RETENTION_POLICY;
+import static be.vlaanderen.informatievlaanderen.ldes.server.retention.services.retentionpolicy.creation.RetentionPolicyConstants.VERSION_BASED_RETENTION_POLICY;
+
+public record TimeAndVersionBasedRetentionPolicy(Duration duration,
+                                                 int numberOfMembersToKeep) implements RetentionPolicy {
+
+    public static TimeAndVersionBasedRetentionPolicy from(RetentionPolicy policyA, RetentionPolicy policyB) {
+        if (policyA instanceof TimeBasedRetentionPolicy timeBasedPolicy) {
+            verifyIsTypeVersionBased(policyB);
+            final int numberOfMembersToKeep = ((VersionBasedRetentionPolicy) policyB).numberOfMembersToKeep();
+            final Duration duration = timeBasedPolicy.duration();
+            return new TimeAndVersionBasedRetentionPolicy(duration, numberOfMembersToKeep);
+        } else if (policyB instanceof TimeBasedRetentionPolicy timeBasedPolicy) {
+            verifyIsTypeVersionBased(policyA);
+            final int numberOfMembersToKeep = ((VersionBasedRetentionPolicy) policyA).numberOfMembersToKeep();
+            final Duration duration = timeBasedPolicy.duration();
+            return new TimeAndVersionBasedRetentionPolicy(duration, numberOfMembersToKeep);
+        } else {
+            throw illegalArgumentException();
+        }
+    }
+
+    private static void verifyIsTypeVersionBased(RetentionPolicy maybeVersionBasedPolicy) {
+        if (!(maybeVersionBasedPolicy instanceof VersionBasedRetentionPolicy)) {
+            throw illegalArgumentException();
+        }
+    }
+
+    private static IllegalArgumentException illegalArgumentException() {
```

Comment:
this seems like a very bad naming.. I understand you want to reuse the exception and that's good. but naming it "illegalargumentException" would make people forget it's actually a function in your class.

Would propose to give the method name a bit more flavour :)

Reply:
I agree and will improve this